app/src/main/java/com/trianguloy/urlchecker/modules/list/WebhookModule.java

```diff
@@ -0,0 +1,191 @@
+
+    public static GenericPref.Str WEBHOOK_URL_PREF(Context cntx) {
+        return new GenericPref.Str("webhook_url", "", cntx);
+    }
+
+    public static GenericPref.Bool AUTO_SEND_PREF(Context cntx) {
+        return new GenericPref.Bool("webhook_auto", false, cntx);
+    }
+
+    @Override
+    public String getId() {
+        return "webhook";
+    }
+
+    @Override
+    public int getName() {
+        return R.string.mWebhook_name;
+    }
+
+    @Override
+    public boolean isEnabledByDefault() {
+        return false;
+    }
+
+    @Override
+    public AModuleDialog getDialog(MainDialog cntx) {
+        return new WebhookDialog(cntx);
+    }
+
+    @Override
+    public AModuleConfig getConfig(ModulesActivity cntx) {
+        return new WebhookConfig(cntx);
+    }
+
+    @Override
+    public List<AutomationRules.Automation<AModuleDialog>> getAutomations() {
+        return (List<AutomationRules.Automation<AModuleDialog>>) (List<?>) WebhookDialog.AUTOMATIONS;
+    }
+}
+
+class WebhookDialog extends AModuleDialog {
+
+    private static final Executor executor = Executors.newSingleThreadExecutor();
+    
+    private final GenericPref.Str webhookUrl;
+    private final GenericPref.Bool autoSend;
+    private TextView statusText;
+    private UrlData currentUrl;
+
+    public static final List<AutomationRules.Automation<WebhookDialog>> AUTOMATIONS = new ArrayList<>();
+    static {
+        AUTOMATIONS.add(new AutomationRules.Automation<>(
+            "webhook_send",
+            R.string.mWebhook_auto_send,
+            dialog -> dialog.sendToWebhook()
+        ));
+    }
+
+    public WebhookDialog(MainDialog dialog) {
+        super(dialog);
+        webhookUrl = WebhookModule.WEBHOOK_URL_PREF(dialog);
+        autoSend = WebhookModule.AUTO_SEND_PREF(dialog);
+    }
+
+    @Override
+    public int getLayoutId() {
+        return R.layout.dialog_webhook;
+    }
+
+    @Override
+    public void onInitialize(View views) {
+        statusText = views.findViewById(R.id.status_text);
+        views.findViewById(R.id.send_button).setOnClickListener(v -> sendToWebhook());
+    }
+
+    @Override
+    public void onDisplayUrl(UrlData urlData) {
+        currentUrl = urlData;
+        if (autoSend.get()) {
+            sendToWebhook();
+        }
+    }
+
+    private void sendToWebhook() {
+        if (currentUrl == null) return;
+        
+        String webhook = webhookUrl.get();
+        if (webhook.isEmpty()) {
+            showStatus(R.string.mWebhook_no_url, true);
+            return;
+        }
+
+        showStatus(R.string.mWebhook_sending, false);
+        
+        executor.execute(() -> {
+            try {
+                JSONObject json = new JSONObject();
+                json.put("url", currentUrl.url);
+                json.put("timestamp", new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US)
+                    .format(new Date()));
+                
+                URL url = new URL(webhook);
```

Comment:
I have a method in HttpUtils that should do this: performPOST
...but it doesn't work.
I guess it's because that method is not setting the content-type nor the requestMethod (apparently virustotal doesn't really care...)

In the virusTotal v3 branch I do have a Connection wrapper that will probably solve this. For now you can leave this code, but could you move it to the HttpUtils class? a `public static boolean performPOSTJson(String url, String body) throws IOException {...}` will suffice